plugin/kubernetes/pvc.go

```diff
@@ -181,28 +199,122 @@ func (v *pvc) runInTemporaryPod(ctx context.Context, cmd []string) ([]byte, erro
 	return bytes, nil
 }
 
+func (v *pvc) runInExistingPod(ctx context.Context, mount mountInfo, cmder makeCommandAtPath) ([]byte, error) {
+	cmd := cmder(mount.path)
+	activity.Record(ctx, "Executing in existing pod: %v", cmd)
+	execRequest := v.client.CoreV1().RESTClient().Post().
+		Resource("pods").
+		Name(mount.pod).
+		Namespace(v.namespace).
+		SubResource("exec").
+		Param("command", cmd[0]).
+		Param("container", mount.container).
+		Param("stderr", "true").
+		Param("stdout", "true")
+
+	for _, arg := range cmd[1:] {
+		execRequest = execRequest.Param("command", arg)
+	}
+
+	executor, err := remotecommand.NewSPDYExecutor(v.config, "POST", execRequest.URL())
+	if err != nil {
+		return nil, err
+	}
+
+	var stdout, stderr bytes.Buffer
+	streamOpts := remotecommand.StreamOptions{
+		Stdout: &stdout,
+		Stderr: &stderr,
+	}
+	err = executor.Stream(streamOpts)
+	activity.Record(ctx, "stdout: %v", stdout.String())
+	activity.Record(ctx, "stderr: %v", stderr.String())
+	if err != nil {
+		return nil, err
+	}
+	return stdout.Bytes(), nil
+}
+
+type mountInfo struct {
+	pod, container, path string
+}
+
+// TODO: return read-write mount if available (fallback to read-only) that mounts the volume
+// root (SubPath is empty). If no mounts exist with empty SubPath, try mounting in a new pod
+// if it's ReadOnly or ReadWriteMany. If that's not possible, error.
+func (v *pvc) getFirstMounter() (*corev1.Pod, string, error) {
+	nsPods, err := v.podi.List(metav1.ListOptions{})
+	if err != nil {
+		return nil, "", err
+	}
+
+	for _, pod := range nsPods.Items {
+		for _, vol := range pod.Spec.Volumes {
+			if vol.PersistentVolumeClaim != nil && vol.PersistentVolumeClaim.ClaimName == v.Name() {
+				return &pod, vol.Name, nil
```

Comment:
Because I'm getting the name of the PersistentVolume object, not the PersistentVolumeClaim object. We need to match that to where it's mounted to a path in a specific container (see getMount).

Reply:
Ah ok. A comment explaining that would help.